bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/source/LineNumberRulerColumn.java

```diff
@@ -687,24 +689,27 @@ private void doubleBufferPaint(GC dest) {
 		}
 
 		if (fBuffer == null) {
-			fBuffer= newFullBufferImage(size);
+			newFullBufferImage(size);
 		} else {
 			doPaint(visibleLines, size);
 		}
 		dest.drawImage(fBuffer, 0, 0);
 	}
 
-	private Image newFullBufferImage(Point size) {
-		ImageGcDrawer imageGcDrawer= (gc, imageWidth, imageHeight) -> {
-			ILineRange lines= JFaceTextUtil.getVisibleModelLines(fCachedTextViewer);
-			if (lines == null) {
```

Comment:
Why is this removed? both `doPaint()` variants will run into NPE if lines == null

Reply:
The problem with this indirection is that if you open call hierarchy to understand who passes what you will see the call here that passes the data without checks, so you either assume it is not checked or you have to go one level deeper etc. Also next maintainer who will look into will assume it is safe not to check the result because it wasn't checked here.

What about to avoid second (expensive) call to `getVisibleModelLines()` and pass the visible lines from the caller to `newFullBufferImage()` instead of calling it yet another time there?

Reply:
Sure, makes to only retrieve the value once anyway (both from a conceptual/comprehensibility as well as from a performance perspective). I adapted the PR to just retrieve the value once and pass it to the method in 4aefad1.